src/lib/blueprint.ts

```diff
@@ -49,6 +55,15 @@ interface Request {
 
 interface Response {
   description: string
+  responseType: 'resource' | 'resource_list' | 'void'
+  responseKey: string | null
+  resourceType: string | null
+}
```

Comment:
For a little nicer type dx, we could do something like

```suggestion
type Response = VoidResponse | ResourceResponse | ResourceListResponse

interface BaseResponse {
  description: string
}

interface VoidResponse extends BaseResponse {
  description: string
  responseType: 'void'
}

interface ResourceResponse extends BaseResponse {
  description: string
  responseType: 'resource'
  responseKey: string 
  resourceType: string
}

interface ResourceListResponse extends BaseResponse {
  description: string
  responseType: 'resource_list'
  responseKey: string
  resourceType: string
}
```